pkg/resource/rule/hooks_tags.go

```diff
@@ -0,0 +1,142 @@
+package rule
+
+import (
+	"context"
+
+	"github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws-controllers-k8s/runtime/pkg/util"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+
+	svcapitypes "github.com/aws-controllers-k8s/eventbridge-controller/apis/v1alpha1"
+)
+
+// getTags retrieves a resource list of tags.
+func (rm *resourceManager) getTags(
+	ctx context.Context,
+	resourceARN string,
+) (tags []*svcapitypes.Tag, err error) {
+	rlog := log.FromContext(ctx)
+	exit := rlog.Trace("rm.getTags")
+	defer func() { exit(err) }()
+
+	var listTagsResponse *eventbridge.ListTagsForResourceOutput
+	listTagsResponse, err = rm.sdkapi.ListTagsForResourceWithContext(
+		ctx,
+		&eventbridge.ListTagsForResourceInput{
+			ResourceARN: &resourceARN,
+		},
+	)
+	rm.metrics.RecordAPICall("GET", "ListTagsForResource", err)
+	if err != nil {
+		return nil, err
+	}
+	for _, tag := range listTagsResponse.Tags {
+		tags = append(tags, &svcapitypes.Tag{
+			Key:   tag.Key,
+			Value: tag.Value,
+		})
+	}
+	return tags, nil
+}
+
+// syncRuleTags updates event bus tags
+func (rm *resourceManager) syncRuleTags(
```

Comment:
I'd recommend just calling this method `syncTags` instead of `syncRuleTags`, since you're already in the `rule` package.

Reply:
Yup, missed those as per your comments on the same situation with `EventBus`